### PR TITLE
 [To rel/13] [IOTDB-4332] data backup tool based on iotdb-session

### PR DESCRIPTION
**Pull request by jeck1son2**
A tool for import and export data from iotdb,including timeseries structure and those data.

-Iotdb-session based
-Pipeline-based import and export tool
-Project-Reactor based
-support more situations,such as export and import a storage,export and import the timeseries structure,the device's align properties.
-support export the data when the query result timeseries > 1000

More information: docs/UserGuide/Maintenance-Tools/Backup-Tool.md

### CODE REVIEW

File: backup-tool/readme_zh.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+````
+-  backup-export.bat -h 127.0.0.1 -p 6667 -u root -pw root -f d:\\validate_test\\83 -i root.** -sy true -se true -c gzip  
+-  backup-import.1.bat -h 127.0.0.1 -p 6667 -u root -pw root -f D:\validate_test\83 -se true -c gzip
+- -h  // iotdb host address
+- -p  // port
+- -u  // username
+- -pw // password
+- -f  // fileFolder
+- -sy // 是否需要时间序列结构,如果是true的话，工具会额外导出一个文件来记录它对应的时间序列结构
+- -se // 文件生成策略，1.以entity的路径为为文件名生成文件 2、额外提供一个文件来记录文件名与entity路径的对应关系
```
Comment (qiaojialin) [inline backup-tool/readme_zh.md:57]: what is entity?

File: backup-tool/readme_zh.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+- -c  // 压缩格式 : SQL、CSV、SNAPPY、GZIP、LZ4
+- -w  // where  条件
+- -i  // 要导出的路径
+````
```
Comment (qiaojialin) [inline backup-tool/readme_zh.md:61]: Give some example

File: backup-tool/pom.xml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            <dependency>
+                <groupId>org.apache.iotdb</groupId>
+                <artifactId>iotdb-antlr</artifactId>
+                <version>${iotdb.version}</version>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+    <dependencies>
+        <!--        <dependency>-->
+        <!--            <groupId>org.slf4j</groupId>-->
```
Comment (qiaojialin) [inline backup-tool/pom.xml:54]: remove useless code

File: docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# Backup-Tool
+IOTDB 的数据导入导出工具,导出的数据主要包含时间序列的结构和其对应的数据。
```
Comment (qiaojialin) [inline docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md:23]: When adding new pages, remember to update the site/src/main/.vuepress in master branch.

File: docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+-  backup-export.bat -h 127.0.0.1 -p 6667 -u root -pw root -f d:\\validate_test\\83 -i root.** -sy true -se true -c gzip  
+-  backup-import.1.bat -h 127.0.0.1 -p 6667 -u root -pw root -f D:\validate_test\83 -se true -c gzip
+- -h  // iotdb host address
+- -p  // port
+- -u  // username
+- -pw // password
+- -f  // fileFolder
+- -sy // 是否需要时间序列结构,如果是true的话，工具会额外导出一个文件来记录它对应的时间序列结构
+- -se // 文件生成策略，1.以entity的路径为为文件名生成文件 2、额外提供一个文件来记录文件名与entity路径的对应关系
+- -c  // 压缩格式 : SQL、CSV、SNAPPY、GZIP、LZ4
```
Comment (qiaojialin) [inline docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md:58]: why we combine the SQL, CSV with SNAPPY?
SQL and CSV is the data structure, SNAPPY is a compresson method, are they in the same level?

File: docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+````
+-  backup-export.bat -h 127.0.0.1 -p 6667 -u root -pw root -f d:\\validate_test\\83 -i root.** -sy true -se true -c gzip  
+-  backup-import.1.bat -h 127.0.0.1 -p 6667 -u root -pw root -f D:\validate_test\83 -se true -c gzip
+- -h  // iotdb host address
+- -p  // port
+- -u  // username
+- -pw // password
+- -f  // fileFolder
+- -sy // 是否需要时间序列结构,如果是true的话，工具会额外导出一个文件来记录它对应的时间序列结构
+- -se // 文件生成策略，1.以entity的路径为为文件名生成文件 2、额外提供一个文件来记录文件名与entity路径的对应关系
```
Comment (qiaojialin) [inline docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md:57]: These parameters, which are used in import and which are used in export?

File: backup-tool/backup-core/src/main/java/org/apache/iotdb/backup/core/DeleteStarter.java [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  public Double[] rateOfProcess() {
+    return new Double[0];
+  }
+
+  @Override
+  public Long finishedRowNum() {
+    return null;
+  }
+
+  // public static void main(String[] args) throws IoTDBConnectionException, InterruptedException {
```
Comment (qiaojialin) [inline backup-tool/backup-core/src/main/java/org/apache/iotdb/backup/core/DeleteStarter.java:125]: remove

File: backup-tool/backup-command/src/main/java/org/apache/iotdb/backup/command/ExportTool13_1.java [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+import org.apache.iotdb.rpc.IoTDBConnectionException;
+import org.apache.iotdb.rpc.StatementExecutionException;
+
+import org.apache.commons.cli.*;
+import reactor.core.Disposable;
+
+import java.io.File;
+
+/** @Author: LL @Description: @Date: create in 2022/7/7 14:29 */
+public class ExportTool13_1 extends AbstractCsvTool {
```
Comment (qiaojialin) [inline backup-tool/backup-command/src/main/java/org/apache/iotdb/backup/command/ExportTool13_1.java:36]: what is 13_1, 0.13.1?
No need to show these in class name, the minor version do not change the TsFile and IoTDB structure/interface.
So only put these code in rel/0.13 is enough.

File: backup-tool/readme_zh.md (hunk 1 above)
Comment (jeck1son2) [reply backup-tool/readme_zh.md:57]: I got the concept of entity from the iotdb official website.
Official description: An entity(Also called device) is an equipped with measurements in real scenarios.
url: https://iotdb.apache.org/UserGuide/Master/Data-Concept/Data-Model-and-Terminology.html#measurement-entity-storage-group-path

File: docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md (hunk 5 above)
Comment (jeck1son2) [reply docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md:58]: There is no combination of SQL, CSV and SNAPPY here; they are just an export file format.
In the way of SQL export, all the exported data is organized in SQL form, you can directly open the file for viewing, and at the same time, you can easily copy several pieces of data and import them directly in the command line client of iotdb. The CSV export method is similar, and the exported data is organized in CSV format for easy viewing.
Other export formats, such as SNAPPY, GZIP, etc.; the data in it is in a compressed format, users cannot view it, and it occupies a small space, so it is more suitable for data backup.

File: backup-tool/readme_zh.md (hunk 2 above)
Comment (jeck1son2) [reply backup-tool/readme_zh.md:61]: add example in readme.md

File: backup-tool/pom.xml (hunk 3 above)
Comment (jeck1son2) [reply backup-tool/pom.xml:54]: removed

File: docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md (hunk 4 above)
Comment (jeck1son2) [reply docs/zh/UserGuide/Maintenance-Tools/Backup-Tool.md:23]: Does that mean I need to submit a special pr to do these changes？

File: backup-tool/backup-core/src/main/java/org/apache/iotdb/backup/core/DeleteStarter.java (hunk 7 above)
Comment (jeck1son2) [reply backup-tool/backup-core/src/main/java/org/apache/iotdb/backup/core/DeleteStarter.java]: removed

File: backup-tool/backup-command/src/main/java/org/apache/iotdb/backup/command/ExportTool13_1.java (hunk 8 above)
Comment (jeck1son2) [reply backup-tool/backup-command/src/main/java/org/apache/iotdb/backup/command/ExportTool13_1.java]: already modified

**Comment by jeck1son2** [comment]
为iotdb-backup-tool单开了一个仓库，此pr作废

**Pull request closed without merging**